core/DataTable/Filter/GroupBy.php

```diff
@@ -104,6 +100,10 @@ public function filter($table)
             }
         }
 
+        if ($this->groupByColumn === 'label') {
+            $table->setLabelsHaveChanged();
```

Comment:
If we change the label column, we need to make sure the dataTableLabelToRowIdIndex will be updated. Otherwise `getRowFromLabel()` won't work.